assets/js/blocks-registry/payment-methods/payment-method-config.js

```diff
@@ -39,10 +42,24 @@ export default class PaymentMethodConfig {
 			showSaveOption: config?.supports?.showSaveOption || false,
 			features: config?.supports?.features || [ 'products' ],
 		};
-		this.canMakePayment = canMakePaymentWithFeaturesCheck(
-			config.canMakePayment,
+		this.canMakePaymentFromConfig = config.canMakePayment;
+	}
+
+	get canMakePayment() {
+		const canPay = canMakePaymentWithFeaturesCheck(
+			this.canMakePaymentFromConfig,
 			this.supports.features
 		);
+
```

Comment:
I think we'd benefit from describing the purpose of the next statement. You can figure it out by reading, but this makes it a bit quicker :)

```suggestion
// Loop through all calbacks to check if there are any registered for this payment method.
```